sdk/azcore/example_test.go

```diff
@@ -81,7 +81,11 @@ func ExampleHTTPResponse() {
 		} else {
 			// ...
 		}
+	} else if err != nil {
+		// Handle non-HTTP error
+		fmt.Printf("There was a non-HTTP error that occurred: %s", err.Error())
```

Comment:
You'll want to avoid that fall-through - in the cases we were brainstorming I'm guessing 'resp' is nil.